queries.sql

```diff
@@ -21,3 +23,96 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+-- VET CLINIC DATABASE: QUERY AND UPDATE ANIMALS TABLE
+
+-- Set species column to unspecified then rollback changes, verify changes
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+
+-- Add species type to the species column
+BEGIN;
+
+UPDATE animals SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+-- Delete all records in the animals table, then roll back the transaction
+BEGIN;
+
+TRUNCATE TABLE animals;
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+
+-- Using savepoint command
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SELECT * FROM animals;
+
+SAVEPOINT sp1;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+SELECT * FROM animals;
+
+ROLLBACK TO sp1;
+
+SELECT * FROM animals;
+
+UPDATE animals SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+SELECT * FROM animals;
+
+COMMIT; 
+
+
+-- How many animals are there?
+SELECT COUNT(*) AS number_of_animals FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) AS never_tried_to_escaped FROM animals
+WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts) AS total_escape_attempts FROM animals
+GROUP BY neutered ORDER BY total_escape_attempts DESC;
+-- neutered animals escape most
```

Comment:
- [ ] For this query, I think for you to determine who escape most; it's either you use the AVG or MAX commands in order to get an accurate result.  ℹ️ 
```suggestion
-- Who escapes the most, neutered or not neutered animals?
SELECT neutered, MAX(escape_attempts) AS total_escape_attempts FROM animals
GROUP BY neutered ORDER BY total_escape_attempts DESC;
-- neutered animals escape most
```

Reply:
<img width="1512" alt="last" src="https://github.com/newhoteng/Vet-clinic/assets/106236756/6eefd2eb-c059-4176-9937-536132d9064a">